tests/test_user_repository.py

```diff
@@ -117,3 +117,45 @@ def test_save_user_unique_violation(
 
     assert exc_info.value.status_code == status.HTTP_409_CONFLICT
     assert "User already exists" in str(exc_info.value.detail)
+
+def test_get_user_by_id_success(user_repository, mock_db_pool, mock_db_connection, mock_db_cursor):
+    # Simulate a valid user returned from the database
+    mock_db_cursor.fetchone.return_value = {
+        "id": 1,
+        "username": "Ann",
+        "email": "ann.aj@example.com",
+        "first_name": "ann",
+        "last_name": "aj",
+        "phone_number": "94872",
+        "role": "user",
+        "status": "active",
+        "address_id": 1,
+        "last_login_at": "2024-01-01",
+        "created_at": "2024-01-01",
+        "updated_at": "2024-01-01",
+    }
```

Comment:
This value can be stored in tests/test_data.py and the variable can be imported to keep the test classes short and clean.

Reply:
Changed